minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/service/MiNiFiConfigurationChangeListener.java

```diff
@@ -0,0 +1,215 @@
+
+                try {
+                    String confDir = bootstrapProperties.getProperty(CONF_DIR_KEY);
+
+                    try {
+                        logger.info("Performing transformation for input and saving outputs to {}", confDir);
+                        ByteBuffer tempConfigFile = generateConfigFiles(newConfigBais, confDir, bootstrapFileProvider.getBootstrapProperties());
+                        runner.getConfigFileReference().set(tempConfigFile.asReadOnlyBuffer());
+
+                        try {
+                            logger.info("Reloading instance with new configuration");
+                            restartInstance();
+                        } catch (Exception e) {
+                            logger.debug("Transformation of new config file failed after transformation into Flow.xml and nifi.properties, reverting.");
+                            try (FileInputStream swapConfigFileStream = new FileInputStream(swapConfigFile)) {
+                                ByteBuffer resetConfigFile = generateConfigFiles(swapConfigFileStream, confDir, bootstrapFileProvider.getBootstrapProperties());
+                                runner.getConfigFileReference().set(resetConfigFile.asReadOnlyBuffer());
+                            }
+                            throw e;
+                        }
+                    } catch (Exception e) {
+                        logger.debug("Transformation of new config file failed after replacing original with the swap file, reverting.");
+                        try (FileInputStream swapConfigFileStream = new FileInputStream(swapConfigFile)) {
+                            Files.copy(swapConfigFileStream, configFile.toPath(), REPLACE_EXISTING);
+                        }
+                        throw e;
+                    }
+                } catch (Exception e) {
+                    logger.debug("Transformation of new config file failed after swap file was created, deleting it.");
+                    if (!swapConfigFile.delete()) {
+                        logger.warn("The swap file failed to delete after a failed handling of a change. It should be cleaned up manually.");
+                    }
+                    throw e;
+                }
+            }
+        } catch (ConfigurationChangeException e){
+            logger.error("Unable to carry out reloading of configuration on receipt of notification event", e);
+            throw e;
+        } catch (IOException ioe) {
+            logger.error("Unable to carry out reloading of configuration on receipt of notification event", ioe);
+            throw new ConfigurationChangeException("Unable to perform reload of received configuration change", ioe);
+        } finally {
+            try {
+                if (configInputStream != null) {
+                    configInputStream.close() ;
+                }
+            } catch (IOException e) {
+                // Quietly close
+            }
```

Comment:
Using one of the commons-lang3 utilities should provide support for a quite close without the empty catch.